dev-lang/golang/golang-1.4.3.recipe

```diff
@@ -0,0 +1,80 @@
+SUMMARY="The Go Programming Language"
+DESCRIPTION="The Go Programming Language is an open source project to make \
+programmers more productive.
+
+Go is expressive, concise, clean, and efficient. Its concurrency \
+mechanisms make it easy to write programs that get the most out of multicore \
+and networked machines, while its novel type system enables flexible and \
+modular program construction. Go compiles quickly to machine code yet has the \
+convenience of garbage collection and the power of run-time reflection. It's a \
+fast, statically typed, compiled language that feels like a dynamically typed, \
+interpreted language."
+HOMEPAGE="https://golang.org/go"
+COPYRIGHT="2009-2018 The Go Authors"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://dl.google.com/go/go$portVersion.src.tar.gz"
+CHECKSUM_SHA256="9947fc705b0b841b5938c48b22dc33e9647ec0752bae66e50278df4f23f64959"
+SOURCE_DIR="go"
+PATCHES="golang-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 !x86 ?x86_64"
+SECONDARY_ARCHITECTURES="!x86"
+
+PROVIDES="
+	golang$secondaryArchSuffix = $portVersion
+	cmd:go$secondaryArchSuffix = $portVersion
+	cmd:gofmt$secondaryArchSuffix = $portVersion
+	cmd:godoc$secondaryArchSuffix = $portVersion
+	"
+if [ "$secondaryArchSuffix" != "" ]; then
+	PROVIDES="$PROVIDES
+	cmd:go = $portVersion
+	cmd:gofmt = $portVersion
+	cmd:godoc = $portVersion
+	"
+fi
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:git
+	"
+
+BUILD()
+{
+	echo "$portVersion" > VERSION
+	export GOROOT_FINAL=/system/develop/lib$secondaryArchSubDir/go
+	cd src
+	DISABLE_ASLR=1 ./make.bash
+	cd ..
+	export GOPATH=/tmp
+	export GOROOT=`pwd`
+}
+
+INSTALL()
+{
+	mkdir -p $developLibDir/go/
+	cp -r src pkg doc $developLibDir/go/
+	mkdir -p $binDir
+	cp bin/* $binDir/
+	if [ "$secondaryArchSuffix" != "" ]; then
```

Comment:
You might wish to use this variant:
```diff
- 	if [ "$secondaryArchSuffix" != "" ]; then
+	if [ -n "$secondaryArchSuffix" ]; then
```

Reply:
Hmm, haikuporter takes care of creating the symlinks (`bin/foobar-x86 -> x86/foobar`), so you should probably drop this bloc. If you don't want to have `foobar-x86` in `bin/` then you should probably define the [commandBinDir](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#using-the-commandbindir-user-variable) User Variable.
```diff
-	if [ "$secondaryArchSuffix" != "" ]; then
-		cd $binDir/..
-		for a in `ls $binDir`
-		do
-			ln -s .$secondaryArchSubDir/$a $a
-		done
- 	fi
```